src/core/controller.ts

```diff
@@ -190,29 +198,29 @@ export class Controller {
       return this;
     }
 
-    let answer = message;
-    if ('parser' in bot.types[rule.type]) {
-      answer = await bot.types[rule.type].parser(answer, rule, bot);
-    }
-
+    let answer;
+    const { executors } = bot.types[rule.type];
+    const executor = executors[this.getRuleExecutorIndex(rule)] || {};
```

Comment:
put in a variable to be reused below
`executorIdx`

Reply:
This is an intended re-check because `transform` and `validate` receives `bot` and then they can change the store...

Reply:
humm okay